packages/backend/src/open_payments/connection/service.ts

```diff
@@ -44,19 +46,8 @@ function getConnection(
       scale: payment.asset.scale
     }
   })
-  return Connection.fromPayment({
-    payment,
-    credentials,
-    openPaymentsUrl: deps.openPaymentsUrl
-  })
-}
-
-function getConnectionUrl(
-  deps: ServiceDependencies,
-  payment: IncomingPayment
-): string | undefined {
-  if (!payment.connectionId) {
-    return undefined
+  return {
```

Comment:
we can probably get away with the `if (!payment.connectionId)` check, and remove `connectionid` altogether from the incoming payment model

Reply:
I removed the `connectionId` from `IncomingPayment`. I kept this check to return undefined when payment is completed or expired to maintain the behavior tested here: 

https://github.com/interledger/rafiki/blob/4f1c1787d2efc14108a56344b42de055590aeafb/packages/backend/src/open_payments/connection/service.test.ts#L93-L107

A `beforeInsert` hook ensured the `connectionId` was always set, and then it would be set to `null` in a `beforeUpdate` hook when the state was completed or expired. So it seems this was basically a roundabout way to check if the payment was expired or complete.

Reply:
Wouldn't this mean that the Open Payments GET /incoming-payment/ fails if the payment is expired or completed since we won't be returning the methods array?

Reply:
In this case, if there is no `ilpStreamConnection` passed into the `incomingPayment.toOpenPaymentsType`, we can just return an empty methods array

Reply:
> Wouldn't this mean that the Open Payments GET /incoming-payment/ fails if the payment is expired or completed since we won't be returning the methods array?

I think you're right but I dont think we can just add an empty methods array to `toOpenPaymentsType` when no connections are passed in. Sometimes it should return the payment without methods to satisfy the base `incoming-payment` openapi type. I guess we could make a seperate `toOpenPaymentsTypeWithMethods` (not overloaded) and call as appropriate. But should the method really be empty when the payment is expired/complete? Do we not want the actual payment method with credentials?

I'm thinking your original suggestion makes the most sense now. If we dont return `undefined` for completed or expired payments then we can handle completed/expired where its called if we want, and do nothing otherwise (such as `GET /incoming-payment/{id}`). I implemented that in this new commit https://github.com/interledger/rafiki/pull/2054/commits/8396970b59a2483728d210d8623d1f2e977b34bb

Reply:
updated. The caller of `toOpenPaymentsType` has to distinguish between which type (with methods or without) so that we ensure we're returning the right object with respect to the methods field. We can no longer use the presence of the `ilpStreamCredentials` to determine it alone. An undefined `ilpStreamCredentials` should sometimes return no methods, and sometimes `[]` depending on where its called.

As such I removed the overloading (rather than perhaps adding an additional `withMethods` boolean) since it makes the internal function logic much simpler and the caller already has to differentiate between which type it wants anyways.